### PR TITLE
Merged structure and bindings from `leptos-leaflet`, converted API to `snake_case`

### LINKED ISSUE #23: Merge `leaflet` from `leptos-leaflet`
Hi there,

[leptos-leaflet](https://github.com/emirror-de/leptos-leaflet/tree/main) has in my opinion a clearer structure and more bindings available than this crate.

Are you interested in merging the changes into your crate? `leptos-leaflet` could then rely on `leaflet-rs` instead of having a separate fork.

If you are happy to merge, I could set up a pull request for it if you do not find the time to do.

Best regards,
Lewin

### PR DESCRIPTION
**Pull request by emirror-de**
Fixes #23

### CODE REVIEW

File: examples/yew-component/dist/leaflet-rs-example-yew-component-1c49ba35ce874fff.js
Comment (flosse) [inline examples/yew-component/dist/leaflet-rs-example-yew-component-1c49ba35ce874fff.js:1]: Compiled files should not be tracked.

File: examples/yew-component/dist/leaflet-rs-example-yew-component-1c49ba35ce874fff_bg.wasm
Comment (flosse) [inline examples/yew-component/dist/leaflet-rs-example-yew-component-1c49ba35ce874fff_bg.wasm:1]: Compiled files should not be tracked.

File: examples/yew-component/dist/style-8f6e4d7b2e92f54.css
Comment (flosse) [inline examples/yew-component/dist/style-8f6e4d7b2e92f54.css:1]: Compiled files should not be tracked.

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "leaflet"
-version = "0.2.3"
-authors = ["slowtec GmbH <post@slowtec.de>"]
+version = "0.3.0"
```
Comment (flosse) [inline Cargo.toml:3]: Incrementing the version should be done in a separate release commit.

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ wrapper for
 
 ```toml
 [dependencies]
-leaflet = "0.2"
+leaflet = "0.3"
```
Comment (flosse) [inline README.md:11]: Incrementing the version should be done in a separate release commit.

**Review by flosse — CHANGES_REQUESTED** [review]
Great work. Just resolve the tiny issues. Thanks!

File: Cargo.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+authors = ["slowtec GmbH <post@slowtec.de>", "Lewin Probst <info@emirror.de>"]
```
Comment (flosse) [inline Cargo.toml:4]: Maybe the original authors of leptos-leaflet should also get some credits here or in the readme.

Comment (emirror-de) [reply Cargo.toml:4]: Yes of course! I was not able to find details about the author. I already asked about discord and am waiting for response. As soon as I do have a response I will update it.

Sorry that I just added it to `Cargo.toml` hopefully you are ok with that. If you prefer adding it to the `README` I am happy to change that as well.

Comment (emirror-de) [reply Cargo.toml:4]: @dgsantana what do you think?

Comment (dgsantana) [reply Cargo.toml:4]: @emirror-de , Hi, I'm ok with having my name and email added in the authors of Cargo.toml if that is ok?
Daniel Santana <danielgsantana@gmail.com>

Comment (dgsantana) [reply Cargo.toml:4]: If we can get feature parity with the leptos-leaflet "forked" leaflet-rs I'm more then happy to just use this as a dependency.

Comment (flosse) [reply Cargo.toml]: Thank you both @emirror-de and @dgsantana!

**Pull request merged**